### PR TITLE
Extended coverage of TensorFlow layer tests

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Added ArgMin, ArgMax tests
 - Added All, Any tests
 - Added FloorDiv, AddV2 to BinaryOps
 - Added IsFinite, Square to UnaryOps
 - Modified AvgPool/MaxPool tests

### Tickets:
 - 92744, 92745, 92746, 92747, ...

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        #
+        #   Create reference IR net
+        #   Please, specify 'type': 'Input' for input node
+        #   Moreover, do not forget to validate ALL layer attributes!!!
+        #
+
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py:57]: please remove it. It is redundant and useless comment.

File: tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            op_type_to_tf = {
+                'ArgMax': tf.raw_ops.ArgMax,
+                'ArgMin': tf.raw_ops.ArgMin,
+            }
+            tf_input_shape = input_shape.copy()
+            tf_input_shape = permute_nchw_to_nhwc(tf_input_shape, use_new_frontend)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py:43]: Not sure that this is needed. Please check. I think it should work without it,

File: tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        ref_net = None
+
+        return tf_net, ref_net
+
+    test_data = [
+        dict(input_shape=[5], dimension=0),            #Simple test of vector
+        dict(input_shape=[2, 3], dimension=1),         #Simple test
+        dict(input_shape=[2, 3, 3, 4], dimension=2),   #Simple test with possible nchw/nhcw
+    ]
+
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py:67]: Please add tf_fe mark to switch it for TF FE in the pre-commit

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            op_type_to_tf = {
+                'All': tf.raw_ops.All,
+                'Any': tf.raw_ops.Any,
+            }
+            tf_input_shape = input_shape.copy()
+            tf_input_shape = permute_nchw_to_nhwc(tf_input_shape, use_new_frontend)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py:49]: Please check I think this is redundant

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        #
+        #   Create reference IR net
+        #   Please, specify 'type': 'Input' for input node
+        #   Moreover, do not forget to validate ALL layer attributes!!!
+        #
+
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py:63]: Please remove it.

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        ref_net = None
+
+        return tf_net, ref_net
+
+    test_data = [
+        dict(input_shape=[2, 3], axis=[1]),         #Simple test
+        dict(input_shape=[2, 3, 3, 4], axis=[1]),   #Simple test with possible nchw/nhwc
+    ]
+
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py:72]: please add test for the pre-commit for TF FE

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        #   Moreover, do not forget to validate ALL layer attributes!!!
+        #
+
+        ref_net = None
+
+        return tf_net, ref_net
+
+    test_data = [
+        dict(input_shape=[2, 3], axis=[1]),         #Simple test
+        dict(input_shape=[2, 3, 3, 4], axis=[1]),   #Simple test with possible nchw/nhwc
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py:70]: please check different `axis`

**Review by rkazants — CHANGES_REQUESTED** [review]

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py (hunk 4 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py]: Done

File: tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py (hunk 2 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py]: Done

File: tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py (hunk 1 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py]: Done

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py (hunk 5 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py]: Done

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py (hunk 7 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py]: Done

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py (hunk 6 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py:64]: Done

File: tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py (hunk 3 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_ArgMinMax.py:59]: Done

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            op_type_to_tf = {
+                'All': tf.raw_ops.All,
+                'Any': tf.raw_ops.Any,
+            }
+            tf_input_shape = input_shape.copy()
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py:45]: please do not create a copy of the shape, it is not needed, remove this in the next PR.

**Review by rkazants — COMMENTED** [review]
Please take a look at the layer tests results. Something is failing.

**Comment by rkazants** [comment]
@gkrivor, is this PR ready? If yes, please resolve comments and re-request review.

File: tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    # use_new_frontend - common parameter
+    def create_logical_ops_placeholder_const_net(self, input_shape, axis, op_type, ir_version, use_new_frontend):
+        """
+            Tensorflow net                  IR net
+
+            Placeholder->op_type   =>       Placeholder->ReduceLogicalAnd/Or
+                         /                               /
+            Const-------/                   Const-------/
+
+        """
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_LogicalOps.py:33]: I propose do not have this scheme, if the operation is placed one-by-one. This looks useless. Please clean-up it in the next PR.

**Review by rkazants — APPROVED** [review]

**Pull request merged**